scipy/interpolate/tests/test_bsplines.py

```diff
@@ -429,7 +430,54 @@ def test_neg_axis(self):
         spl0 = BSpline(t, c[0], k)
         spl1 = BSpline(t, c[1], k)
         assert_equal(spl(2.5), [spl0(2.5), spl1(2.5)])
-
+    
+    def test_design_matrix_bc_types(self):
+        np.random.seed(1234)
+        n = 11
+        k = 3
+        bc_list = ['natural', 'not-a-knot','clamped','periodic']
```

Comment:
When it's a classmethod, boundary conditions are not relevant anymore, so let's restructure the test